### DISCUSSION TITLE
python memory leak when using CPU execution provider with multiple threads

### DISCUSSION THREAD

**Discussion by JustinNober** [microsoft/onnxruntime#16898] [Other Q&A]
When I use onnxruntime-cpu to use multithreading on python, I found the problem of memory growth; in my continuous loop test reasoning, I found that the free memory in the cpu is getting smaller and smaller.

![图片](https://github.com/microsoft/onnxruntime/assets/69416588/dad7cc9d-4707-44b3-8406-9827221eb15f)

The same problem occurs in the threading library。

The red box is the memory I monitor，This gets smaller and smaller and doesn't release.

![图片](https://github.com/microsoft/onnxruntime/assets/69416588/5c300a1f-49a6-4f88-a37d-fa7e6bec9120)

**Comment by RyanUnderhill** [comment]
Have you tried forcing garbage collection periodically and seeing what effect that has?

**Reply by JustinNober** [reply]
Tried but nothing works.

**Comment by snnn** [comment]
Maybe there really has a memory leak, but the way you watch memory usage is wrong. Do not look at the "free" column". The last column is what really matters.

**Reply by JustinNober** [reply]
thanks
